frontend/src/app/modules/leader/leader-board/leader-board.component.ts

```diff
@@ -56,19 +90,131 @@ export class LeaderBoardComponent extends BaseComponent implements OnInit {
+        }
+
+        this.friendsPage.pageNumber++;
+        this.loading = true;
+
+        this.userService
+            .getFriendsLeaderBoard(this.friendsPage)
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe({
+                next: (users) => {
+                    this.handleFriendsUpdate(users);
                 },
                 error: () => {
                     this.loading = false;
                     this.toastrNotification.showError('Server connection error');
                 },
             });
     }
+
+    private getCurrentUser() {
+        this.userService
+            .getCurrentUser()
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe({
+                next: (user) => {
+                    this.currentUser = user;
+                    this.userFriendsIds = user.friendships.map((f) => f.friendId);
+                },
+                error: () => {
+                    this.toastrNotification.showError('Server connection error');
+                },
+            });
+    }
+
+    public isCurrentUser(user: IUser): boolean {
+        return user && this.currentUser && user.id === this.currentUser.id;
+    }
+
+    public isAddFriendButtonVisible(user: IUser): boolean {
+        return !this.isCurrentUser(user) && !this.isFriend(user);
+    }
+
+    public isPendingFriendship(user: IUser): boolean {
+        return !this.isCurrentUser(user) && this.getFriendshipStatus(user) === FriendshipStatus.Pending;
+    }
+
+    public isAcceptedFriendship(user: IUser): boolean {
+        return !this.isCurrentUser(user) && this.getFriendshipStatus(user) === FriendshipStatus.Accepted;
+    }
+
+    private getFriendshipStatus(user: IUser): FriendshipStatus | undefined {
+        return this.currentUser?.friendships?.find((f) => f.friendId === user.id)?.friendshipStatus;
+    }
+
+    public isFriend(user: IUser): boolean {
+        return user && this.userFriendsIds.includes(user.id);
+    }
+
+    public addFriend(user: IUser) {
+        const request: INewFriendship = {
+            senderId: this.currentUser.id,
+            recipientId: user.id,
+        };
+
+        this.userService
+            .sendFriendshipRequest(request)
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe({
+                next: (userDto) => {
+                    this.currentUser.friendships = userDto.friendships;
+                    this.userFriendsIds = userDto.friendships.map((f) => f.friendId);
+                },
+                error: (error) => {
+                    this.toastrNotification.showError(error);
+                },
+            });
+    }
+
+    public toggleMyFriends() {
+        this.isMyFriendsChecked = !this.isMyFriendsChecked;
+
+        if (this.isMyFriendsChecked) {
+            this.friendsPage.pageNumber = 0;
+            this.isLastFriendsPage = false;
+            this.getFriends();
+        } else {
+            this.usersToShow = this.users;
+        }
```

Comment:
not clear what is happening here

Reply:
when user switch "my friends", we send request, preliminary updating the page number (infinite scroll setting) in order to get first page of out friends. In other case we just show users leaderboard

Reply:
see the previous comments